src/util/std_types.h

```diff
@@ -995,10 +1002,12 @@ inline const code_typet &to_code_type(const typet &type)
 */
 inline code_typet &to_code_type(typet &type)
 {
-  PRECONDITION(type.id()==ID_code);
+  PRECONDITION(can_cast_type<code_typet>(type));
+  validate_type(type);
```

Comment:
Heh - never heard of this before - any reason not to stick this in the `can_cast_type`?

Reply:
If this fails it doesn't mean that the type is not the right type, it means that there's an internal inconsistency.